ParabolInc/parabol
# feat(recurrence): Added advanced recurrence settings

# Description

Fixes #6922, #7497, #7496

Changes introduced in this PR reflect technical spec discussed [here.](https://www.notion.so/parabol/Technical-specification-3787e836c2d946b3a52789c4ea6d4c44#43248b21271d4aea821413b5c198b84b)
- Removed `startRecurrence` and `stopRecurrence` mutation
- Introduced `updateRecurrenceSettings` mutation that handles both scenarios 
- Migrated `startTeamPrompt` mutation to SDL 
- Introduced `RRule` gql scalar type
- A lot of UI

TODO:

- [x] Handle `tzid` issue
- [ ] Migration for making `duration` nullable - will be handled in separate PR

## Demo

https://www.loom.com/share/09082094a1e14e168e6af28fe3d1d12a

## Testing scenarios

- [ ] Enabling recurrence from the new meeting view
  - Open new meeting modal
  - Select standup
  - Enable recurrence by selecting various options from the Recurrence Settings dropdown
  - Start the meeting, see if recurrence is active with correct options

- [ ] Updating recurrence from the existing meeting view
  - Prepare a meeting with recurrence enabled (first scenario)
  - Open meeting options menu in the top right corner, select Edit Recurrence Settings
  - Edit the recurrence settings, try different options
  - Click update, see if recurrence is correctly updated
  - Make sure a snackbar is displayed after updating recurrence settings

- [ ] Disabling recurrence from the existing meeting view
  - Prepare a meeting with recurrence enabled (first scenario)
  - Open meeting options menu in the top right corner, select Edit Recurrence Settings
  - Uncheck all the days 
  - Click update, see if recurrence was correctly disabled
  - Make sure a snackbar is displayed after disabling recurrence

- [ ] Enable recurrence from the meeting view
  - Start new standup meeting without recurrence enabled
  - Open meeting options menu in the top right corner, select Start Recurrence
  - Select the recurrence settings
  - Click update, see if recurrence was correctly started
  - Make sure a snackbar is displayed after updating recurrence settings

## Final checklist
